packages/dataviews/src/view-table.tsx

```diff
@@ -459,16 +468,10 @@ function ViewTable< Item >( {
 		setNextHeaderMenuToFocus( fallback?.node );
 	};
 
-	const viewFields = view.fields || fields.map( ( f ) => f.id );
-	const visibleFields = fields.filter(
-		( field ) =>
-			viewFields.includes( field.id ) ||
-			[ view.layout.mediaField ].includes( field.id )
-	);
 	const hasData = !! data?.length;
-
-	const primaryField = fields.find(
-		( field ) => field.id === view.layout.primaryField
+	const fieldRenderConfigs = normalizeFieldRenderConfigs(
+		view.fields,
+		fields
 	);
```

Comment:
```suggestion
	const fieldsToRender = getFieldsToRender(
		view.fields,
		fields
	);
```

How would you like this proposal? I find it makes the code self-explanatory from here on.

Reply:
For me `fieldsToRender` and `view.fields` are actually just the same thing but in order to avoid manipulating multiple supported shapes, we're normalizing it with this function and adding default values in case view.fields is empty. 

I don't mind the change of name if it was the only place where this was used but we also have `normalizeFields` function which does the exact same thing but for `fields`.

Also I don't like the `fieldsToRender` variable because these are not fields, they are "rendering configurations for fields".

Reply:
I see your point. I am coming from a different angle ([see](https://github.com/WordPress/gutenberg/pull/63148#discussion_r1666512418)). It's ok to ship this as it is until we fully clarify the role of the renderConfig/renderAs property.